### PR TITLE
Update to 4.0.0 for new Rest API

### PR DESCRIPTION
**Pull request by bgklein**
- Added the functionality to get which version of the Azure Batch Node Agent (https://github.com/Azure/Batch/blob/master/changelogs/nodeagent/CHANGELOG.md) is running on nodes via the NodeAgentInformation property of ComputeNode.
 - Added additional error handling to task add operations to prevent deadlock scenarios.
 - Removed validation status from count operations.
 - Updated adal, azure-client-runtime, and commons-codec dependencies.
 - Update 'List' methods to return the lower layer PagedList object instead of the standard List.

### CODE REVIEW

File: src/main/java/com/microsoft/azure/batch/TaskOperations.java [hunk 1]
```diff
@@ -123,7 +151,8 @@ public Exception getException() {
         /**
          * Submits one chunk of tasks to a job.
          * 
-         * @param taskList A list of {@link TaskAddParameter tasks} to add.
+         * @param taskList
+         *            A list of {@link TaskAddParameter tasks} to add.
          */
         private void submit_chunk(List<TaskAddParameter> taskList) {
```
Comment (xingwu1) [inline src/main/java/com/microsoft/azure/batch/TaskOperations.java]: Remove the 'private' modifier for the private variable of private class workingThread.

File: src/main/java/com/microsoft/azure/batch/protocol/implementation/AccountsImpl.java [hunk 2]
```diff
@@ -298,7 +296,7 @@ public AccountsImpl(Retrofit retrofit, BatchServiceClientImpl client) {
     /**
      * Lists all node agent SKUs supported by the Azure Batch service.
      *
-     * @param accountListNodeAgentSkusOptions Additional parameters for the operation
+    ServiceResponseWithHeaders<PageImpl<NodeAgentSku>, AccountListNodeAgentSkusHeaders> * @param accountListNodeAgentSkusOptions Additional parameters for the operation
```
Comment (xingwu1) [inline src/main/java/com/microsoft/azure/batch/protocol/implementation/AccountsImpl.java]: This comment is due to autorest tool bug. Please remove them.

File: README.md [hunk 3]
```diff
@@ -2,12 +2,12 @@
 
 ## Azure Batch Libraries for Java
 
-This README is based on the latest released version Azure Batch SDK (3.1.0). If you are looking for other releases, see the [More Information](#more-information) section below.
+This README is based on the latest released version Azure Batch SDK (3.4.0). If you are looking for other releases, see the [More Information](#more-information) section below.
```
Comment (xingwu1) [inline README.md]: Did you update the pom.xml to update the dependency version?

Comment (xingwu1) [reply README.md]: At least I couldn't find it.

**Comment by bgklein** [comment]
Ping @xingwu1. Updated to reflect comments.

File: README.md [hunk 4]
```diff
@@ -2,12 +2,12 @@
...
 
 The Azure Batch Libraries for Java is a higher-level, object-oriented API for interacting with the Azure Batch service.
 
 
-> **3.2.0** is a release that supports all features of Azure Batch service with API version "2018-03-01.6.1". We will be adding support for more new features and tweaking the API associated with Azure Batch service newer release.
+> **3.4.0** is a release that supports all features of Azure Batch service with API version "2018-08-01.7.1". We will be adding support for more new features and tweaking the API associated with Azure Batch service newer release.
```
Comment (xingwu1) [inline README.md]: Change to 4.0.0 for the version.

File: changelog.md [hunk 5]
```diff
@@ -1,5 +1,16 @@
 # Azure Batch SDK for Java release notes
 
+## Changes in 3.4.0
```
Comment (xingwu1) [inline changelog.md]: Same here.

**Pull request merged**